aiohttp/http_parser.py

```diff
@@ -225,6 +225,11 @@ def parse_headers(
         return (CIMultiDictProxy(headers), tuple(raw_headers))
 
 
+def _is_supported_upgrade(headers: CIMultiDictProxy[str]) -> bool:
+    """Check if the upgrade header is supported."""
+    return headers.get(hdrs.UPGRADE, "").lower() in ("tcp", "websocket")
```

Comment:
`tcp` is needed here or `test_upgrade_connection_not_released_after_read` failed. Are there any others?